src/Api.Rest.Dtos/JsonConverters/JsonDataCharacteristicArrayConverter.cs

```diff
@@ -0,0 +1,96 @@
+﻿#region copyright
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+/* Carl Zeiss IMT (IZfM Dresden)                   */
+/* Softwaresystem PiWeb                            */
+/* (c) Carl Zeiss 2015                             */
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#endregion
+
+namespace Zeiss.PiWeb.Api.Rest.Dtos.JsonConverters
+{
+	#region usings
+
+	using System;
+	using System.Buffers;
+	using System.Buffers.Text;
+	using System.Collections.Generic;
+	using System.Text.Json;
+	using System.Text.Json.Serialization;
+	using Zeiss.PiWeb.Api.Rest.Dtos.Data;
+
+	#endregion
+
+	/// <summary>
+	/// Specialized <see cref="JsonConverter"/> for <see cref="DataCharacteristicDto"/>-objects.
+	/// </summary>
+	public sealed class JsonDataCharacteristicArrayConverter : JsonConverter<IReadOnlyCollection<DataCharacteristicDto>>
+	{
+		#region methods
+
+		/// <inheritdoc />
+		public override IReadOnlyCollection<DataCharacteristicDto> Read( ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options )
+		{
+			var characteristics = new List<DataCharacteristicDto>();
+
+			while( reader.Read() && reader.TokenType == JsonTokenType.PropertyName )
+			{
+				var uuidSpan = reader.HasValueSequence ? reader.ValueSequence.ToArray() : reader.ValueSpan;
+
+				if( !Utf8Parser.TryParse( uuidSpan, out Guid uuid, out var bytesConsumed ) || uuidSpan.Length != bytesConsumed )
+				{
+					throw new FormatException( $"Input span was not in a correct format, on converting to '{typeof( Guid ).Name}'" );

```

Comment:
You could also use the easier nameof( Guid ) in the error message.

Reply:
Fixed